assets/src/edit-story/components/library/panes/media/media3p/api/media3pApiProvider.js

```diff
@@ -0,0 +1,141 @@
+ */
+function Media3pApiProvider({ children }) {
+  const MEDIA_PAGE_SIZE = 20;
+
+  function constructFilter(provider, searchTerm, mediaType) {
+    let tokens = [];
+    if (provider) {
+      tokens.push(`provider:${provider}`);
+    }
+    if (mediaType) {
+      tokens.push(`type:${mediaType}`);
+    }
+    if (searchTerm) {
+      tokens.push(searchTerm);
+    }
+    return tokens.join(' ');
+  }
+
+  function getUrls(m) {
+    if (m.type.toLowerCase() === 'image') {
+      return Object.fromEntries(
+        new Map(m.imageUrls.map((u) => [u.imageName, u.url]))
+      );
+    }
+    throw new Error('Invalid media type.');
+  }
+
+  function getFullAsset(m) {
+    if (m.type.toLowerCase() === 'image') {
+      return m.imageUrls.find((i) => i.imageName === 'full');
+    }
+    throw new Error('Invalid media type.');
+  }
+
+  /**
+   * Get media for the given parameters.
+   *
+   * @param {Object} obj - An object with the options.
+   * @param {string} obj.provider The provider to get the media from. Currently only 'unsplash' is supported.
+   * @param {?string} obj.searchTerm Optional search term to send, eg: 'cute cats'.
+   * @param {?string} obj.orderBy The desired ordering of the results. Defaults to 'relevance' in the API.
+   * @param {?string} obj.mediaType The media type of results to get. Currently ignored by the API as Unsplash only handles images.
+   * @param {?string} obj.pageToken An optional page token to provide, for pagination.
+   * @return {Promise<{nextPageToken: *, media: *}>} An object with the media resources and a next page token.
+   */
+  async function listMedia({
+    provider,
+    searchTerm,
+    orderBy,
+    mediaType,
+    pageToken,
+  }) {
+    if (provider.toLowerCase() !== Providers.UNSPLASH) {
+      throw new Error(`Unsupported providers: ${provider}`);
+    }
+    const response = await apiListMedia({
+      filter: constructFilter(provider, searchTerm, mediaType),
+      orderBy: orderBy,
+      pageSize: MEDIA_PAGE_SIZE,
+      pageToken: pageToken,
+    });
+    return {
+      media: response.media.map((m) => {
+        const urls = getUrls(m);
+        const fullAsset = getFullAsset(m);
+        return createResource({
+          type: m.type.toLowerCase(),
+          mimeType: fullAsset.mimeType,
+          creationDate: m.createTime,
+          src: fullAsset.url,
+          width: fullAsset.width,
+          height: fullAsset.height,
+          poster: null, // TODO: Implement for videos.
+          posterId: null, // TODO: Implement for videos.
+          id: m.name,
+          length: null, // TODO: Implement for videos.
+          lengthFormatted: null, // TODO: Implement for videos.
+          title: m.description,
+          alt: null,
+          local: false, // TODO: How does this interact with the rest?
+          sizes: urls, // TODO: Map with expected keys for canvas.
+        });
+      }),
+      nextPageToken: response.nextPageToken,
+    };
+  }
+
+  const state = {
```

Comment:
The term 'state' can be confused for the 'state' attribute that exists in many context values. Consider renaming to 'value' or 'contextValue'.